### PR TITLE
refactor api

### PR DESCRIPTION
**Pull request by EtienneLt**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

#300

### CODE REVIEW

File: pypowsybl/flowdecomposition/impl/__init__.py
Comment (colineplqt) [inline pypowsybl/flowdecomposition/impl/__init__.py:1]: Why is this necessary?

File: tests/test_security_analysis.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def test_security_analysis_parameters():
     sa.add_single_element_contingency('NHV1_NHV2_2', 'First contingency')
     sa.add_postcontingency_monitored_elements(branch_ids=['NHV1_NHV2_1'], contingency_ids='First contingency')
 
-    # default security analysis
+    # default sensitivity analysis
```
Comment (colineplqt) [inline tests/test_security_analysis.py]: Is it?

File: pypowsybl/flowdecomposition/impl/__init__.py
Comment (EtienneLt) [reply pypowsybl/flowdecomposition/impl/__init__.py:1]: to use the methods outside the package they need to be exposed in the __init__.py file

File: tests/test_security_analysis.py (hunk 1 above)
Comment (EtienneLt) [reply tests/test_security_analysis.py]: oups

File: pypowsybl/loadflow/impl/validation_result.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+from typing import Optional
```
Comment (OpenSuze) [inline pypowsybl/loadflow/impl/validation_result.py:7]: missing header

File: pypowsybl/network/impl/layout_parameters.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import pypowsybl._pypowsybl as _pp
```
Comment (OpenSuze) [inline pypowsybl/network/impl/layout_parameters.py:7]: missing header

File: pypowsybl/network/impl/network_creation_util.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
+import io
```
Comment (OpenSuze) [inline pypowsybl/network/impl/network_creation_util.py:7]: missing header

File: pypowsybl/network/impl/node_breaker_topology.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
+from pandas import DataFrame
```
Comment (OpenSuze) [inline pypowsybl/network/impl/node_breaker_topology.py:7]: missing header

File: pypowsybl/loadflow/impl/loadflow.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
+#
```
Comment (OpenSuze) [inline pypowsybl/loadflow/impl/loadflow.py]: remove this first line

File: pypowsybl/security/impl/security.py [hunk 7]
```diff
@@ -0,0 +1,138 @@
+#
```
Comment (OpenSuze) [inline pypowsybl/security/impl/security.py]: remove first line

File: pypowsybl/sensitivity/impl/sensitivity.py [hunk 8]
```diff
@@ -0,0 +1,138 @@
+#
```
Comment (OpenSuze) [inline pypowsybl/sensitivity/impl/sensitivity.py]: remove first line and add SPDX identifier

File: pypowsybl/flowdecomposition/impl/flowdecomposition.py [hunk 9]
```diff
@@ -1,106 +1,26 @@
 #
-# Copyright (c) 2020, RTE (http://www.rte-france.com)
+# Copyright (c) 2023, RTE (http://www.rte-france.com)
```
Comment (colineplqt) [inline pypowsybl/flowdecomposition/impl/flowdecomposition.py:2]: Do you have to change the year here?

Comment (EtienneLt) [reply pypowsybl/flowdecomposition/impl/flowdecomposition.py:2]: I dont know, better to be the most recent no ?

**Review by geofjamg — APPROVED** [review]
